### ISSUE TITLE
Check that media-consent & ticket generation works on patience

### ISSUE THREAD

**Issue by PeterJCLaw** [srobo/tasks#61]
Since `patience` (our current volunteer-services machine) is considerably newer than `saffron` (where these were previously hosted), we should check that the generation of media consent and ticket PDFs actually works before we tell teams to use them.

The approach here would be to run up a dev instance of [`patience`](https://github.com/PeterJCLaw/sr-server-puppet/tree/patience) and go through the motions of generating & submitting a media consent form and then a ticket against that dev instance.

**Comment by PeterJCLaw** [comment]
This is on hold while we work out whether we're actually going to deploy media consent and/or tickets this year.

**Comment by RealOrangeOne** [comment]
This is now unblocked. See https://github.com/srobo/tasks/issues/59#issuecomment-466913870

**Comment by PeterJCLaw** [comment]
Once I realised that it wasn't working for me because I wasn't in a college (and worked out how to see that error message), it turns out that the media consent side of this is already working.
I'd not anticipated that volunteers can't use this mechanism, but I think that's a limitation we can cope with.

I've not checked ticket generation yet, will do that now.

**Comment by PeterJCLaw** [comment]
Hrm, the tickets side doesn't seem to work quite right for me locally (even as a competitor account).

**Comment by PeterJCLaw** [comment]
Ah, the tickets stuff contains the dates hard-coded:
``` php
    exec("cd tickets ; python generate.py $user -y 2017 -d 'April 1st-2nd (Doors open 9:00)' -l 'Event details: http://goo.gl/Rk1I7d' -o $ticket_name 2>&1", $output, $rv);
```
https://www.studentrobotics.org/cgit/ticket-access.git/tree/get-ticket.php#n12

We should probably fix that.

**Comment by PeterJCLaw** [comment]
Ok, looks like the tickets system needs a slight tweak to work on PHP 7 (where `$HTTP_RAW_POST_DATA` doesn't exist.

**Comment by PeterJCLaw** [comment]
I've also just realised that it doesn't get the proper URL baked into it, which means that it'll always end up querying the live data, which would make local testing rather hard. We _sooo_ need to rework this system (after this competition, when we have more time).

**Comment by PeterJCLaw** [comment]
The next error is that the TLS cert is invalid, which of course it isn't (as I'm on a dev box). Solving that shows another HTTP 500 coming from `horrendous.php`.
I'm calling it a night for now, but here's the patches I've applied to get this far:

``` diff
[root@sr-vm tickets]# git diff
diff --git a/user_database.py b/user_database.py
index 14d4f72..e9b91bc 100644
--- a/user_database.py
+++ b/user_database.py
@@ -2,7 +2,7 @@ import collections, json
 
 with open('ticket.key', 'r') as f:
     _key = f.read()
-_BASE_URI = "https://studentrobotics.org/ticket-api/horrendous.php"
+_BASE_URI = "http://localhost/ticket-api/horrendous.php"
 
 class User:
     _shared_users = {}
@@ -15,11 +15,14 @@ class User:
     def get(self, username):
         if username not in self._shared_users:
             user = self(username)
-            import urllib2
+            import urllib2, ssl
+            ctx = ssl.create_default_context()
+            ctx.check_hostname = False
+            ctx.verify_mode = ssl.CERT_NONE
             request = urllib2.Request(_BASE_URI,
                                       self._encode_query({'username': username}),
                                       {'Content-type': 'application/octet-stream'})
-            f = urllib2.urlopen(request)
+            f = urllib2.urlopen(request, context=ctx)
             raw_data = f.read()
             print raw_data
             data = json.loads(raw_data)
diff --git a/webapi/horrendous.php b/webapi/horrendous.php
index 3088d0e..928ad4f 100644
--- a/webapi/horrendous.php
+++ b/webapi/horrendous.php
@@ -21,7 +21,7 @@ function college_name_from_group($ldap_connection, $group) {
 
 $config = parse_ini_file('config.ini');
 
-$data = $HTTP_RAW_POST_DATA;
+$data = file_get_contents('php://input');
 if (!$data) {
     deny('No data.');
 }
```

**Comment by trickeydan** [comment]
> We _sooo_ need to rework this system (after this competition, when we have more time).

I'd like to volunteer to help with this. I have some ideas about phasing out LDAP for next year also. It's based off some work that I did for SUSU last year (albeit never deployed by them :( ). Happy to discuss whenever, but I suggest when we have more time and less other things to do.

**Comment by RealOrangeOne** [comment]
Discussion for future iterations of the media-consent forms and related process should happen on https://github.com/srobo/tasks/issues/60

**Comment by PeterJCLaw** [comment]
Ok, I'm not sure what other issues I was hitting previously, however having applied the fixes discovered in the patch above (plus merging some other latent fixes which were in the repos), this now works for me locally.

**Comment by PeterJCLaw** [comment]
After some beating, this is now working.
